src/assets/js/themes.js

```diff
@@ -1,49 +1,48 @@
 /* theme toggle buttons */
-(function() {
-    var enableToggle = function(btn) {
+(function () {
+    var enableToggle = function (btn) {
         btn.setAttribute("aria-pressed", "true");
     }
-
-    var disableToggle = function(btn) {
+    var disableToggle = function (btn) {
         btn.setAttribute("aria-pressed", "false");
     }
+    var setTheme = function (theme) {
+        document.documentElement.setAttribute('data-theme', theme);
+        window.localStorage.setItem("theme", theme);
+    }
 
+    var theme = window.localStorage.getItem("theme");
+    if (!theme || (theme !== "light" && theme !== "dark")) {
+        var isDark = window.matchMedia('(prefers-color-scheme: dark)').matches;
+        theme = isDark ? "dark" : "light";
+    }
 
-    let theme = window.localStorage.getItem("theme");
-
-    document.addEventListener('DOMContentLoaded', function() {
+    document.addEventListener('DOMContentLoaded', function () {
         var switcher = document.getElementById('js-theme-switcher');
         switcher.removeAttribute('hidden');
-
         var light_theme_toggle = document.getElementById('light-theme-toggle'),
             dark_theme_toggle = document.getElementById('dark-theme-toggle');
-
-
-        if (theme == "light") {
+        if (theme === "light") {
             enableToggle(light_theme_toggle);
             disableToggle(dark_theme_toggle);
-        } else if (theme == "dark") {
+            setTheme(theme);
+        } else if (theme === "dark") {
             enableToggle(dark_theme_toggle);
             disableToggle(light_theme_toggle);
+            setTheme(theme);
```

Comment:
I think we don't need this part of the code too... 
- https://github.com/eslint/new.eslint.org/blob/7bd0f8f4c3be64337f2392203cf952d8a512143b/src/_includes/layouts/base.html#L68-L73

During initial page load this method actually setstheme as dark mode if the system preference is dark else we dont set any theme.